src/Task.ts

```diff
@@ -47,15 +47,30 @@ export class Task {
     /** The blockLink is a "^" annotation after the dates/recurrence rules. */
     public readonly blockLink: string;
 
-    public static readonly dateFormat = 'YYYY-MM-DD';
     public static readonly taskRegex = /^([\s\t]*)[-*] +\[(.)\] *(.*)/u;
     // The following regexes end with `$` because they will be matched and
     // removed from the end until none are left.
     public static readonly priorityRegex = /([⏫🔼🔽])$/u;
-    public static readonly startDateRegex = /🛫 ?(\d{4}-\d{2}-\d{2})$/u;
-    public static readonly scheduledDateRegex = /[⏳⌛] ?(\d{4}-\d{2}-\d{2})$/u;
-    public static readonly dueDateRegex = /[📅📆🗓] ?(\d{4}-\d{2}-\d{2})$/u;
-    public static readonly doneDateRegex = /✅ ?(\d{4}-\d{2}-\d{2})$/u;
+
+    public static readonly dateRegex = '\\d{4}-?\\d{2}-?\\d{2}';
+    public static readonly dateOrLinkedDateRegex = `(${Task.dateRegex}|\\[\\[${Task.dateRegex}\\]\\])`;
+
+    public static readonly startDateRegex = new RegExp(
+        `🛫 ?${Task.dateOrLinkedDateRegex}$`,
```

Comment:
Instead of duplicating the emojis around that are use for the tasks add in constants and then just refer to that, then you can refer to the same constant through all the code to make maintenance simpler.

```
public static readonly defaultStartDateToken = '🛫';

 public static readonly startDateRegex = new RegExp(
        `${defaultStartDateToken} ?${Task.dateOrLinkedDateRegex}$`,
        'u',
    );

public static readonly defaultDueDateToken = '📅';
public static readonly optionalDueDateTokenOne = '📆';
public static readonly optionalDueDateTokenOne = '🗓';
```

Reply:
Will do, thank you for the idea! There’s lots of duplication of many things at the moment 🙈